app/routes.py

```diff
@@ -1 +1,275 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, abort, make_response, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import os
+import requests
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"details":"Invalid data"}, 400))
+
+    model = cls.query.get(model_id)
+    
+    if not model:
+        abort(make_response({"details": f"{cls.__name__} {model_id} is not found"}, 404))
+    
+    return model
+
+def validate_request(cls, request):
+    request_body = request.get_json()
+
+    if not request_body.get("title"):
+        abort(make_response({"details":"Invalid data"}, 400))
+    
+    elif not request_body.get("description") and cls == Task:
+        abort(make_response({"details":f"Invalid data"}, 400))
+    
+    return request_body
+  
+
+@task_bp.route("", methods=["POST"])
+def create_task():
+    request_body = validate_request(Task, request)
+
+    new_task = Task.from_dict(request_body)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task":new_task.to_dict()}, 201
+
+@task_bp.route("", methods=["GET"])
+def read_all_tasks():
+
+    sort_query = request.args.get("sort", "asc")
+    title_query = request.args.get("title")
+    description_query = request.args.get("description")
+   
+
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+
+    elif title_query:
+        tasks = Task.query.filter_by(title = title_query)
+
+    elif description_query:
+        tasks = Task.query.filter_by(description = description_query)
+
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+    return jsonify(tasks_response), 200
+   
+
+@task_bp.route("/<id>", methods=["GET"])
+def read_single_task(id):
+    task = validate_model(Task, id)
+
+    goal = Goal.query.get(id)
+
+    if goal != None:
+        task_dict = task.to_dict()
+        task_dict["goal_id"] = goal.goal_id
+
+    else:
+        task_dict = task.to_dict()
+
+    return ({"task":task_dict}),200
+    
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title  = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
Right now, if a user sends a request without the keys title or description your server would crash. There's a couple of ways to handle this, you could call the `validate_request` function before you access the keys in `request_body` or you could implement a `try/except` block.